### PR TITLE
Bookstore: Initialize project components

### PR DESCRIPTION
**Pull request by Mukumbuta**
### Features implemented:
- Set up React with the create-react-app command
- Set up linters
- Ste up React `BrowserRouter`.
- Set up `UUID `for dynamic generation of book IDs
- Added React Router and set two `Route `s and `Link `s for the app's navigation: `Books `and `Categories
`
- Made Books route as the default view
- Added functionality to display the list of books (empty at this point but it is ready for the data) with the Remove button (no functionality yet)
- Created a component called Book for displaying a single book (receive a title and an author as props)
- Reused the Book component in the component that displays a list of books
- Added a form component for adding a book (with inputs for a title and an author)
-
 Categories route
 displays a button "Check status" only.

### CODE REVIEW

File: src/components/DisplayBookList.js [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+        {bookDetails.map((book) => (
+          <Book key={book.id} books={book} />
+        ))}
+      </ul>
+    </div>
+  );
+};
+
+DisplayBookList.propTypes = {
+  // eslint-disable-next-line react/forbid-prop-types
```
Comment (arslanbisharat) [inline src/components/DisplayBookList.js]: - [ ] Please make sure you are not disabling linter rules. These are here to help us write clean code.

File: src/index.css [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+body {
+  font-family: cursive;
+}
+
+nav {
+  display: flex;
+  justify-content: space-between;
+  align-items: center;
+
+  /* background: #371300; */
```
Comment (arslanbisharat) [inline src/index.css]: - [ ] Please make sure there is no commented code. This makes hard to read the code.

File: src/index.css [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+  margin-inline: 100px;
+  height: 80px;
+}
+
+nav h1 {
+  font-size: 35px;
+
+  /* color: #28a99e; */
```
Comment (arslanbisharat) [inline src/index.css]: - [ ] Please make sure there is no commented code. This makes hard to read the code.

File: src/index.css [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+ul {
+  display: flex;
+  gap: 10px;
+}
+
+.nav-links {
+  font-weight: 600;
+  font-size: 17px;
+
+  /* border-right: 3px solid hsl(202, 11%, 29%); */
```
Comment (arslanbisharat) [inline src/index.css]: - [ ] Please make sure there is no commented code. This makes hard to read the code.

File: src/index.css [hunk 5]
```diff
@@ -0,0 +1,66 @@
...
+}
+
+.nav-links:last-of-type {
+  border-right: none;
+}
+
+/* .active-link {
+  color: orangered;
+  text-decoration: underline;
+} */
```
Comment (arslanbisharat) [inline src/index.css]: - [ ] Please make sure there is no commented code. This makes hard to read the code.

File: src/components/Book.js [hunk 6]
```diff
@@ -0,0 +1,22 @@
...
+        <span>{books.title}</span>
+        <span>{books.author}</span>
+        <button type="button" className="remove-btn">Remove</button>
+      </li>
+    </div>
+  );
+};
+
+Book.propTypes = {
+  // eslint-disable-next-line react/forbid-prop-types
```
Comment (arslanbisharat) [inline src/components/Book.js]: - [ ]  Please make sure you are not disabling linter rules. These are here to help us write clean code.

**Review by arslanbisharat — CHANGES_REQUESTED** [review]
Hi @Mukumbuta, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights

- The Pull Request has a proper title and description. ✔️
- There is no linter error is present. ✔️
- Project has met all the technical aspects you needed for this project. Keep rocking! ✔️

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
- [ ] Please make sure you right descriptive comments. Commenting involves placing Human Readable Descriptions inside of computer programs detailing what the Code is doing. Proper use of commenting can make code maintenance much easier, as well as helping make finding bugs faster. Further, commenting is very important when writing functions that other people will use. :x:


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by DammyShittu — APPROVED** [review]
Hi @Mukumbuta,

## APPROVED 🟢

## Highlights
✔️ Descriptive PR
✔️ Passing linters
✔️ Good branch name

Great implementation of all the requirements in this milestone. 👏 

_You have done an amazing job 👍_

Your project is complete!  There is nothing else to say other than... it's time to merge it. :shipit: 

Congratulations! 🎉 

## CHEERS AND HAPPY CODING!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**